java/server/src/co/mitro/core/servlets/GetMyPrivateKey.java

```diff
@@ -221,13 +227,43 @@ private void sendEmailAndThrow(GetMyPrivateKeyRequest in, MitroRequestContext co
       String token = makeLoginTokenString(identity, in.extensionId, in.deviceId);
       String tokenSignature = TwoFactorSigningService.signToken(token);
 
-      if (!context.manager.isReadOnly()) {
-        DBEmailQueue email = DBEmailQueue.makeNewDeviceVerification(identity.getName(), token, tokenSignature, context.platform, context.requestServerUrl);
-        context.manager.emailDao.create(email);
-        context.manager.commitTransaction();
+      GenericRawResults<String[]> queuedEmails = context.manager.emailDao.queryRaw(
+              "SELECT type_string, arg_string FROM ("
+              + "(SELECT type_string, arg_string FROM email_queue)"
+              + " UNION "
+              + "(SELECT type_string, arg_string FROM email_queue_sent) "
+              + ") WHERE type_string=? AND arg_string LIKE ?",
+              DBEmailQueue.Type.LOGIN_ON_NEW_DEVICE.getValue(),
+              "%" + identity.getName() + "%"
+      );
+      boolean recentEmail = false;
+
+      final Date now = new Date(); // this should be UTC
```

Comment:
This is what the documentation says at: http://docs.oracle.com/javase/7/docs/api/java/lang/System.html#currentTimeMillis()